### PR TITLE
[Bugfix] Fixed overflowing category names

### LINKED ISSUE #3631: Text overflow with when editing forum category.
**Describe the bug**
Text overflows if the category name is too long. 

**Expected behavior**
Text should either wrap, or page should alert user that the category name is too long.

**To Reproduce**
1. Go to Discussion Forum
2. Click on create thread
3. Click on Edit Categories
4. Click the edit icon and use insert a big string
5. See error

**Configuration**
 - OS: Linux
 - Browser Firefox/ Chrome

**Screenshots**
overflow

### PR DESCRIPTION
**Pull request by xprilion**
### Please check if the PR fulfills these requirements:

* [x] The PR title and message follows our [guidelines](https://submitty.org/developer/how_to_contribute)

### What is the current behavior?
Fixes #3631 

### What is the new behavior?
Category names are broken into lines depending upon maximum width available. 

https://www.loom.com/share/4cd7b707d79340bcba7dad5712fe413b

Screenshot from 2019-05-04 15-56-50

Screenshot from 2019-05-04 15-59-19

### Other information?
Is this a breaking change?- No, just CSS changes

### CODE REVIEW

**Comment by xprilion** [comment]
Just found out something weird about this...the categories on the Create Thread page displays correctly (as shown above) only for student. Not for the instructor or TA. 

However, the 'View Source' on Chrome displays the changed code for the same. But it renders without that. Very weird, confounded! 

Even tried out commenting out the entire categories list, even then it shows! :o

**Comment by bmcutler** [comment]
I think it would be reasonable to put a character limit on the category.  30 or 50 characters should be plenty.  Anyone else have thoughts?  Extra long categories will cause formatting problems in multiple places.

**Comment by xprilion** [comment]
@bmcutler I guess at times the categories might be more than 30 characters. However, 50 looks like a good upper cap. I vote in for a max limit to the characters here! The user is free to use multiple categories to make their post more specific.

**Comment by MasterOdin** [comment]
I'd agree that 50 is probably more than enough characters for a category name and would be a good max.

**Comment by andrewaikens87** [comment]
I agree that 50 characters should be a good max. When displaying post content in the thread list, we cap at 80 characters (though this is strictly visual). I will re-review once the appropriate changes have been made to cap the length of the categories.

**Comment by xprilion** [comment]
@bmcutler @andrewaikens87 the weird behaviour still continues - the changes on the categories are coming in effect only for the Student profiles. Not for TA or Instructor. However, on view-source of the page, the changes as made are visible on the source. But the rendered output discards them.

**Comment by bmcutler** [comment]
@andrewaikens87 
Can you take a look?  Is it ok to merge as is?

**Comment by xprilion** [comment]
@andrewaikens87 I have made the changes as requested. Kindly note that I have still left the clipping part for backwards compatibility with systems which have long category names already but update Submitty to the latest version. Please let me know if that should be removed!

File: site/app/templates/forum/ThreadPostForm.twig [hunk 1]
```diff
@@ -133,4 +133,17 @@
     $("input[name='title']").change(function () {
         $(this).val($.trim($(this).val()));
     });
+
+    $(".cat-buttons").each(function () {
```
Comment (andrewaikens87) [inline site/app/templates/forum/ThreadPostForm.twig]: I'm not sure we should add `...` for categories that are longer than 50. For legacy categories, that might be greater than 50, I think we should just make sure it doesn't overflow.

File: site/app/templates/forum/ThreadPostForm.twig [hunk 2]
```diff
@@ -133,4 +133,17 @@
     $("input[name='title']").change(function () {
```
[the remark this answers is not in the document]
Comment (xprilion) [reply site/app/templates/forum/ThreadPostForm.twig]: That did solve the problem! Thanks! Could you please let me know why did that work? :o

File: site/app/templates/forum/CategoriesForm.twig [hunk 3]
```diff
@@ -96,6 +96,13 @@
                 refresh_color_select($(this));
             });
         });
+
+        function checkInputMaxLength(obj){
```
[the remark this answers is not in the document]
Comment (xprilion) [reply site/app/templates/forum/CategoriesForm.twig]: Sure, will add that too! (I got confused by an earlier statement and thought all the characters restrictions have to be strictly on the front end only!)

File: site/app/templates/forum/ThreadPostForm.twig (hunk 1 above)
Comment (xprilion) [reply site/app/templates/forum/ThreadPostForm.twig]: Got it, fixing that! :)

**Pull request merged**